linux-user/syscall.c

```diff
@@ -8323,6 +8324,8 @@ static abi_long do_syscall1(void *cpu_env, int num, abi_long arg1,
                 do_sys_futex(g2h(cpu, ts->child_tidptr),
                              FUTEX_WAKE, INT_MAX, NULL, NULL, 0);
             }
+
+            object_unref(OBJECT(cpu));
```

Comment:
Looking at current QEMU, this should be done before unlocking clone_lock

Reply:
the thing is that:

from comments in object_initialize_child_with_propsv():
>   * We want @obj's reference to be 1 on success, 0 on failure.
     * On success, it's 2: one taken by object_initialize(), and one
     * by object_property_add_child().
     * On failure in object_initialize() or earlier, it's 1.
     * On failure afterwards, it's also 1: object_unparent() releases
     * the reference taken by object_property_add_child().

next, when thread done, obj->ref == 2;
We doing just single obj_unref(), which leads to:
`if (qatomic_fetch_dec(&obj->ref) == 1) {`, which executed as `if (2 == 1)`,
and decreasing obj->ref by one, after fetch...
So, the `cpu` object, which is analogue to `task_struct` kernel object, still allocated and forgotten.

Moreover, even if we pass this branch, then in 
`static void object_finalize(void *data)` we have `g_assert(obj->parent == NULL);` which expect no-linkage to parent. 

Finally, we remove linkage to parent while holding a lock, and freeing (unknown to parent now) object without.